### PR TITLE
Newq ds question

### PR DESCRIPTION
**Pull request by shubhamr238**
added my mock interview question.

### CODE REVIEW

File: DS-And-Algo/index.html [hunk 1]
```diff
@@ -153,6 +155,16 @@ <h4>
                     </h4>
                 </a>
             </li>
+             
+
+            <li>
+                <a href="./answers/shubhamr238-q4.html">
+                    <h4>
+                       Puppy Treat (Array)
```
Comment (shivamyth) [inline DS-And-Algo/index.html:175]: Explain question bit here as well.

**Review by shivamyth — CHANGES_REQUESTED** [review]

File: DS-And-Algo/index.html (hunk 1 above)
Comment (shubhamr238) [reply DS-And-Algo/index.html:175]: this was a mock interview question

Comment (shubhamr238) [reply DS-And-Algo/index.html:175]: it is difficult to explain the question there. it's just an array question.

Comment (shubhamr238) [reply DS-And-Algo/index.html:175]: i have explained the whole question in the html file

Comment (shivamyth) [reply DS-And-Algo/index.html:175]: Yeah but still write a bit about the question there as well

**Review by shivamyth — DISMISSED** [review]

**Review by shivamyth — APPROVED** [review]

**Pull request merged**